### PR TITLE
RUMM-140 Add performance benchmarks (3/3)

### PR DESCRIPTION
**Pull request by ncreated**
### What and why?

🎁 This PR adds benchmark tests for `Logging` part.

### How?

Two test cases are created:
* benchmarks for Logger public API;
* benchmarks for `FileWriter` and `FileReader` to measure logging I/O.

#### iPhone X
I made real measurements on physical device - `iPhone X, iOS 13.3.1`. Baselines for this device are included in this PR. At any point, I can run benchmarks locally and see if they pass ✅.

Results from `iPhone X, iOS 13.3.1`:

Screenshot 2020-02-28 at 16 40 23

#### CI
So far, Bitrise test device instrumentation is not configured (it's still beta feature for Bitrise iOS and we may add it in the future: `RUMM-249`). Right now, the CI is only configured to run benchmarks in Simulator, without comparing results against any baselines. This way we ensure benchmarks are working.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference

### CODE REVIEW

File: instrumented-tests/Benchmark/BenchmarkTests/BenchmarkTests.swift [hunk 1]
```diff
@@ -6,28 +6,44 @@
...
+        try! Datadog.deinitializeOrThrow()
         super.tearDown()
     }
 
     // MARK: - `HTTPServerMock` connection
 
-    func connectToServer() throws -> ServerMock {
+    private static func setUpMockServerConnection() throws -> ServerMock {
         let testsBundle = Bundle(for: BenchmarkTests.self)
         guard let serverAddress = testsBundle.object(forInfoDictionaryKey: "MockServerAddress") as? String else {
```
Comment (mariusc83) [inline instrumented-tests/Benchmark/BenchmarkTests/BenchmarkTests.swift]: I missed the part where you get the value from Info.plist...can you point me to this ?

File: instrumented-tests/Benchmark/BenchmarkTests/Logs/LoggingIOBenchmarkTests.swift [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            queue.sync {} // wait to complete async write
+        }
+
+        // Wait enough time for `reader` to accept the youngest batch file
+        Thread.sleep(forTimeInterval: LogsPersistenceStrategy.Constants.minFileAgeForRead + 0.1)
+
+        measureMetrics([.wallClockTime], automaticallyStartMeasuring: false) {
+            self.startMeasuring()
+            let batch = reader.readNextBatch()
+            self.stopMeasuring()
```
Comment (mariusc83) [inline instrumented-tests/Benchmark/BenchmarkTests/Logs/LoggingIOBenchmarkTests.swift]: why not : `measure {}` ?

File: instrumented-tests/Benchmark/BenchmarkTests/Logs/LoggingIOBenchmarkTests.swift [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+            XCTAssertNotNil(batch, "Not enough batch files were created for this benchmark.")
+
+            if let batch = batch {
```
Comment (mariusc83) [inline instrumented-tests/Benchmark/BenchmarkTests/Logs/LoggingIOBenchmarkTests.swift]: Do you think you need this call ?

File: instrumented-tests/Benchmark/BenchmarkTests/BenchmarkTests.swift (hunk 1 above)
Comment (ncreated) [reply instrumented-tests/Benchmark/BenchmarkTests/BenchmarkTests.swift]: Yah, it's quite complex to follow due to lack of fully programmatic build system in Xcode. It's composed of different IDE capabilities.

It works this:
* The "Build Pre-action" (defined [here](https://github.com/DataDog/dd-sdk-ios/pull/35/files#diff-a6d4b9747b42e7454a9ea4407e4ea59a)) executes [generate-http-server-mock-config.sh](https://github.com/DataDog/dd-sdk-ios/blob/ncreated/RUMM-140-add-performance-benchmarks-part-4/tools/config/generate-http-server-mock-config.sh) script:
Screenshot 2020-03-03 at 09 54 04

* This script writes server IP address to `http-server-mock.xcconfig` which is [.gitignored](https://github.com/DataDog/dd-sdk-ios/pull/35/files#diff-12a0eaaff469bf6c651883ed1a91903c). The `http-server-mock.xcconfig` looks this:
```
MOCK_SERVER_ADDRESS=127.0.0.1:8000
```
* `http-server-mock.xcconfig` is imported conditionally by [`tests-specific-config.xcconfig`](https://github.com/DataDog/dd-sdk-ios/pull/35/files#diff-9010028a8f7408ef36a5c8c02685cb2e).
* `tests-specific-config.xcconfig` is used to provide build-time variables to Xcode build system:
Screenshot 2020-03-03 at 10 00 23

* Finally, the `$(MOCK_SERVER_ADDRESS)` variable is used in `Info.plist` dictionary:

Screenshot 2020-03-03 at 10 01 56

So, for each build:
* The IP is generated and written to `xcconfig`;
* From `xcconfig` it is passed to `Info.plist`;
* It is read from `Info.plist` at runtime.

File: instrumented-tests/Benchmark/BenchmarkTests/Logs/LoggingIOBenchmarkTests.swift (hunk 2 above)
Comment (ncreated) [reply instrumented-tests/Benchmark/BenchmarkTests/Logs/LoggingIOBenchmarkTests.swift]: The `measure {}` benchmarks the entire closure and can't measure only part of it. Here, I wanted to isolate and benchmark only `reader.readNextBatch()` operation (not `markBatchAsRead()`), thus I needed `startMeasuring()` + `stopMeasuring()`. It's very similar to what we did for Android benchmarks, so I modelled it the same way, to be able to compare results.

File: instrumented-tests/Benchmark/BenchmarkTests/Logs/LoggingIOBenchmarkTests.swift (hunk 3 above)
Comment (ncreated) [reply instrumented-tests/Benchmark/BenchmarkTests/Logs/LoggingIOBenchmarkTests.swift]: It's this or force unwrapping:
```swift
reader.markBatchAsRead(batch!)
```
which I dislike in tests, as it crashes the entire suite (vs just "failing" this test and continuing with others).

**Pull request merged**